py3status/modules/deadbeef.py

```diff
@@ -67,56 +79,48 @@ def _is_running(self):
 
     def deadbeef(self):
         color = self.color_stopped
-        artist = isplaying = length = playback_time = title = \
-            tracknumber = year = ""
+        status = ''
 
         if not self._is_running():
             return {
                 'cached_until': self.py3.time_in(self.cache_timeout),
-                'color': color,
-                'full_text': ''
+                'full_text': self.py3.safe_format(self.format),
+                'color': color
             }
 
-        # mix metadata fields in with the delimiters
-        fmt = self.delimiter.join([
-            '%artist%', '%isplaying%', '%length%', '%playback_time%',
-            '%title%', '%tracknumber%', '%year%'
-        ])
+        # merge format placeholders with format separators
+        # merge hidden placeholders with control separators
+        # merge strings together to run deadbeef command only once
+        out = FORMAT_SEPARATOR.join(['%{}%'.format(x) for x in self.placeholders])
+        control = [CONTROL_SEPARATOR + s for s in (CONTROL_PARAMETERS)]
+        fmt = ''.join(out) + ''.join(control)
```

Comment:
Simple is always better. `git diff`, please... when you can. I would like to see how it should be done simple/better using same separator because I can't see how I'm making this extra work and complication when it's all done in `post_config_hook` now.

I guess we'll use some loops and pass a number.